File: day2/day2.cpp
/*
    A - Rock
    B - Paper
    C - Scissors

    X - Choose Rock
    Y - Choose Paper
    Z - Choose Scissors

    Points Awarded for Shape
    1  - Rock
    2  - Paper
    3  - Scissors 

    Points Awarded for Outcome
    0 - Lose
    3 - Draw
    6 - Win
*/

#include <iostream>
#include <vector>
#include <string>
#include <fstream>
#include <sstream>

using namespace std;

//class Score() {
//    private:
//        string choice = "";
//        char outcome = '';
//        int score = 0;
//    public:
//        getChoice() { return choice; }
//        getOutcome() { return outcome; }
//        getScore() { return score; }
//}

int tallys[10];

void tallyPointsPart1(string input)
{
    if (input == "B X")
        tallys[1]++;   
    if (input == "C Y")
        tallys[2]++;
    if (input == "A Z")
        tallys[3]++;
    if (input == "A X")
        tallys[4]++;
    if (input == "B Y")
        tallys[5]++;
    if (input == "C Z")
        tallys[6]++;
    if (input == "C X")
        tallys[7]++;
    if (input == "A Y")
        tallys[8]++;
    if (input == "B Z")
        tallys[9]++;
}

void calcScore()
{
    int total_score = 0;
    for (int i = 0; i < 10; i++)
        total_score += tallys[i] * i;
    cout << "Total Score: " << total_score << endl;
}

void getInputVector(string filename)
{
    string input = "";
    ifstream readFromFile(filename);
    while (getline(readFromFile, input)) {
        tallyPointsPart1(input);
    }
}

void clearTallys()
{
    for (int i = 0; i < 10; i++)
        tallys[i] = 0;
}

void printTallys()
{
    for (int i = 0; i < 10; i++)
        cout << i << " " << tallys[i] << endl;
}

int main()
{
    //Part 1
    getInputVector("small_input_day2.txt");
    calcScore();
    clearTallys();
    getInputVector("large_input_day2.txt");
    calcScore();
    clearTallys();

    return 0;
}

File: day2/day2.py
def tallyPointsPart1(input_string, tally_list):
    if input_string == "B X":
        tally_list[1] += 1
    if input_string == "C Y":
        tally_list[2] += 1
    if input_string == "A Z":
        tally_list[3] += 1
    if input_string == "A X":
        tally_list[4] += 1
    if input_string == "B Y":
        tally_list[5] += 1
    if input_string == "C Z":
        tally_list[6] += 1
    if input_string == "C X":
        tally_list[7] += 1
    if input_string == "A Y":
        tally_list[8] += 1
    if input_string == "B Z":
        tally_list[9] += 1
    
def tallyPointsPart2(input_string, tally_list):
    if input_string == "B X":
        tally_list[1] += 1
    if input_string == "C X":
        tally_list[2] += 1
    if input_string == "A X":
        tally_list[3] += 1
    if input_string == "A Y":
        tally_list[4] += 1
    if input_string == "B Y":
        tally_list[5] += 1
    if input_string == "C Y":
        tally_list[6] += 1
    if input_string == "C Z":
        tally_list[7] += 1
    if input_string == "A Z":
        tally_list[8] += 1
    if input_string == "B Z":
        tally_list[9] += 1 

def getList(file_path):
    tally_list = []
    with open(file_path, 'r') as file:
        tally_list = [x.strip() for x in file.readlines()]
        return tally_list

def calcScore(tally_list):
    total_score = 0
    for x in range(len(tally_list)):
        total_score += tally_list[x] * x
    return total_score

def main():
    #Lists that hold input
    small_tally_list = getList("small_input_day2.txt")
    large_tally_list = getList("large_input_day2.txt")
    
    #Part 1
    small_tally_counter = [0] * 10
    large_tally_counter = [0] * 10
    for x in range(len(small_tally_list)):
        tallyPointsPart1(small_tally_list[x], small_tally_counter)
    for x in range(len(large_tally_list)):
        tallyPointsPart1(large_tally_list[x], large_tally_counter)
    print("Total Score: " + str(calcScore(small_tally_counter)))
    print("Total Score: " + str(calcScore(large_tally_counter)))

    #Part2
    small_tally_counter_2 = [0] * 10
    large_tally_counter_2 = [0] * 10
    for x in range(len(small_tally_list)):
        tallyPointsPart2(small_tally_list[x], small_tally_counter)
    for x in range(len(large_tally_list)):
        tallyPointsPart2(large_tally_list[x], large_tally_counter)
    print("Total Score: " + str(calcScore(small_tally_counter_2)))
    print("Total Score: " + str(calcScore(large_tally_counter_2)))


if __name__ == "__main__":
    main()